 neo4j/v1/routing.py | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/neo4j/v1/routing.py b/neo4j/v1/routing.py
index 3fab8b6..14e387f 100644
--- a/neo4j/v1/routing.py
+++ b/neo4j/v1/routing.py
@@ -20,9 +20,13 @@
 
 
 from abc import abstractmethod
-from sys import maxsize
+from sys import maxsize, version_info
 from threading import Lock
-from time import clock
+
+if version_info >= (3, 8):
+    from time import perf_counter
+else:
+    from time import clock as perf_counter
 
 from neo4j.addressing import SocketAddress
 from neo4j.bolt import ConnectionPool, ServiceUnavailable, ProtocolError, DEFAULT_PORT, connect, ConnectionErrorHandler
@@ -85,7 +89,7 @@ class OrderedSet(MutableSet):
 
 class RoutingTable(object):
 
-    timer = clock
+    timer = perf_counter
 
     @classmethod
     def parse_routing_info(cls, records):
