 circuits/web/parsers/multipart.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/circuits/web/parsers/multipart.py b/circuits/web/parsers/multipart.py
index 618e899..0aaf8db 100644
--- a/circuits/web/parsers/multipart.py
+++ b/circuits/web/parsers/multipart.py
@@ -41,7 +41,8 @@ from tempfile import TemporaryFile
 from wsgiref.headers import Headers
 import re
 try:
-    from urlparse import parse_qs
+    # For Python 3.8+; cgi.parse_qs last appeared in 3.7.
+    from urllib.parse import parse_qs
 except ImportError:  # pragma: no cover (fallback for Python 2.5)
     from cgi import parse_qs
 try:
