 variation/redis.py | 44 ++++++++++++++++++++++++++++++++++----------
 1 file changed, 34 insertions(+), 10 deletions(-)

diff --git a/variation/redis.py b/variation/redis.py
index 21609c0..8fe4d10 100644
--- a/variation/redis.py
+++ b/variation/redis.py
@@ -24,6 +24,7 @@ from snmpsim.record.snmprec import SnmprecRecord
 from snmpsim.mltsplit import split
 from snmpsim import error, log
 from pysnmp.smi.error import WrongValueError
+from pyasn1.compat import octets
 
 try:
     from redis import StrictRedis
@@ -80,6 +81,29 @@ def init(**context):
 
 unpackTag = SnmprecRecord().unpackTag
 
+def lindex(dbConn, *args):
+    ret = dbConn.lindex(*args)
+    if ret is not None:
+        ret = octets.octs2str(ret)
+
+    return ret
+
+
+def get(dbConn, *args):
+    ret = dbConn.get(*args)
+    if ret is not None:
+        ret = octets.octs2str(ret)
+
+    return ret
+
+
+def evalsha(dbConn, *args):
+    ret = dbConn.evalsha(*args)
+    if ret is not None:
+        ret = octets.octs2str(ret)
+
+    return ret
+
 
 def variate(oid, tag, value, **context):
     if 'dbConn' in moduleContext:
@@ -112,7 +136,7 @@ def variate(oid, tag, value, **context):
         keySpaceIdx = int((time.time() - moduleContext['booted']) % (recordContext['settings']['period'] * int(dbConn.llen(keySpacesId))) // recordContext['settings']['period'])
     else:
         keySpaceIdx = 0
-    keySpace = dbConn.lindex(keySpacesId, keySpaceIdx)
+    keySpace = lindex(dbConn, keySpacesId, keySpaceIdx)
     if ('current-keyspace' not in recordContext or
             recordContext['current-keyspace'] != keySpace):
         log.msg('redis: now using keyspace %s (cycling period %s)' % (keySpace, recordContext['settings']['period'] or '<disabled>'))
@@ -133,9 +157,9 @@ def variate(oid, tag, value, **context):
             textValue = str(context['origValue'])
 
         if redisScript:
-            prevTagAndValue = dbConn.evalsha(redisScript, 1, keySpace + '-' + dbOid)
+            prevTagAndValue = evalsha(dbConn, redisScript, 1, keySpace + '-' + dbOid)
         else:
-            prevTagAndValue = dbConn.get(keySpace + '-' + dbOid)
+            prevTagAndValue = get(dbConn, keySpace + '-' + dbOid)
         if prevTagAndValue:
             prevTag, prevValue = prevTagAndValue.split('|')
             if unpackTag(prevTag)[0] != unpackTag(textTag)[0]:
@@ -147,23 +171,23 @@ def variate(oid, tag, value, **context):
                            dbOid)
 
         if redisScript:
-            dbConn.evalsha(redisScript, 1, keySpace + '-' + dbOid, textTag + '|' + textValue)
+            evalsha(dbConn, redisScript, 1, keySpace + '-' + dbOid, textTag + '|' + textValue)
         else:
             dbConn.set(keySpace + '-' + dbOid, textTag + '|' + textValue)
 
         return origOid, textTag, context['origValue']
     else:
         if context['nextFlag']:
-            textOid = dbConn.lindex(keySpace + '-oids_ordering',
+            textOid = lindex(dbConn, keySpace + '-oids_ordering',
                                     getNextOid(dbConn, keySpace, dbOid,
                                                index=True))
         else:
             textOid = keySpace + '-' + dbOid
 
         if redisScript:
-            tagAndValue = dbConn.evalsha(redisScript, 1, textOid)
+            tagAndValue = evalsha(dbConn, redisScript, 1, textOid)
         else:
-            tagAndValue = dbConn.get(textOid)
+            tagAndValue = get(dbConn, textOid)
 
         if not tagAndValue:
             return origOid, tag, context['errorStatus']
@@ -182,7 +206,7 @@ def getNextOid(dbConn, keySpace, dbOid, index=False):
     while maxlen >= minlen and listsize:
         listsize -= 1
         idx = minlen + (maxlen - minlen) // 2
-        nextOid = dbConn.lindex(listKey, idx)
+        nextOid = lindex(dbConn, listKey, idx)
         if nextOid < oidKey:
             minlen = idx + 1
         elif nextOid > oidKey:
@@ -192,7 +216,7 @@ def getNextOid(dbConn, keySpace, dbOid, index=False):
             break
     if not listsize:
         raise error.SnmpsimError('empty/unsorted %s' % listKey)
-    return not index and dbConn.lindex(listKey, idx) or idx
+    return not index and lindex(dbConn, listKey, idx) or idx
 
 
 def record(oid, tag, value, **context):
@@ -237,7 +261,7 @@ def record(oid, tag, value, **context):
 
     dbConn.lpush(keySpace + '-temp_oids_ordering', keySpace + '-' + dbOid)
     if redisScript:
-        dbConn.evalsha(redisScript, 1, keySpace + '-' + dbOid, textTag + '|' + textValue)
+        evalsha(dbConn, redisScript, 1, keySpace + '-' + dbOid, textTag + '|' + textValue)
     else:
         dbConn.set(keySpace + '-' + dbOid, textTag + '|' + textValue)
 
